File: frontend/src/app/dtos/event-dto.ts
import {AbstractControl, ValidationErrors} from '@angular/forms';
import {Performance} from './performance/performance';
import {Location} from './location/location';
export class Artist {
  constructor(public data) {}
}

class EventImages {
}

export class EventDto {
  constructor(
    public id: number,
    public title: string,
    public description: string,
    public category:  string,
    public duration: (string | ((control: AbstractControl) => (ValidationErrors | null))[])[],
    public artist: Artist,
    public performances: Performance[],
    public location: Location,
    public eventImages: EventImages[]
  ) {
  }
}

File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/endpoint/dto/events/EventDto.java
package at.ac.tuwien.sepm.groupphase.backend.endpoint.dto.events;

import at.ac.tuwien.sepm.groupphase.backend.entity.enums.Category;

import java.util.List;

public class EventDto extends EventPreviewDto {
    private List<PerformanceDto> performances;
    private LocationDto location;
    private PreviewArtistDto artist;

    public EventDto(int id, String title, String description, Category category, int duration, List<PerformanceDto> performances, LocationDto location, PreviewArtistDto artist) {
        super(id, title, description, category, duration);
        this.performances = performances;
        this.location = location;
        this.artist = artist;
    }

    public EventDto(int id, String name, String description, Category category, int duration) {
        super(id, name, description, category, duration);
    }

    public List<PerformanceDto> getPerformances() {
        return performances;
    }

    public LocationDto getLocation() {
        return location;
    }

    public PreviewArtistDto getArtist() {
        return artist;
    }


}
